core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -589,10 +600,10 @@ public void process(@Nullable MappingMetaData mappingMd, boolean allowIdGenerati
         }
 
         // generate id if not already provided and id generation is allowed
-        if (allowIdGeneration) {
-            if (id == null) {
-                id(UUIDs.base64UUID());
-            }
+        if (allowIdGeneration && id == null) {
+            assert autoGeneratedTimestamp == -1;
+            autoGeneratedTimestamp = System.currentTimeMillis();
```

Comment:
paranoia: should we do `Math.max(0, System.currentTimeMillis())` to protect against the case someone's clock is set before 1970? (It might work to consider the id non automatic, but maybe it's better to just use 0 as a timestamp in that case?)